kgforge/specializations/stores/nexus/prepare_methods.py

```diff
@@ -0,0 +1,188 @@
+import copy
+from typing import Dict, Optional, Tuple, Type
+from typing_extensions import TypeAlias
+from aiohttp import hdrs
+
+from kgforge.core.resource import Resource
+import kgforge.core.commons.exceptions as run_exceptions
+from kgforge.core.conversions.rdf import as_jsonld
+from kgforge.specializations.stores.nexus.service import Service
+
+prepare_return: TypeAlias = Tuple[
+    str, str, Resource, Type[run_exceptions.RunException], Dict, Optional[Dict], Optional[Dict]
+    # method, url, resource, exception to throw, headers, params, payload
+]
+
+# TODO maybe params as a separate arg is not necessary.
+#  Used to be only service parameters being propagated,
+#  but since the service is available, the param can be retrieved here
+
+
+def prepare_create(
+        service: Service,
+        resource: Resource,
+        **kwargs
+) -> prepare_return:
+
+    schema_id = kwargs.get("schema_id")
+
+    params_register = copy.deepcopy(service.params.get("register", {}))
+
+    identifier = resource.get_identifier()
+
+    url = Service.add_schema_and_id_to_endpoint(
+        service.url_resources, schema_id=schema_id, resource_id=identifier
+    )
+
+    context = service.model_context or service.context
```

Comment:
remind me, what did we say was the difference between the `model_context` and the `context`?

Reply:
the questions goes on the lines of ... if we are treating them as "the same", why do we have both? do we need them in different places?

Reply:
I don't know what the difference would be in the sense of use cases of having different contexts for the store and the model, but with respect to the configuration file (yaml), you have a context defined in your store, and you can define one (optionally) in the model. If no model is defined in the model, then the store model value is kept.

Reply:
hmm OK